plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1856,7 +1864,8 @@ void producer_plugin_impl::produce_block() {
    chain.commit_block();
 
    block_state_ptr new_bs = chain.head_block_state();
-   _producer_watermarks[new_bs->header.producer] = chain.head_block_num();
+
+   consider_new_watermark( new_bs->header.producer, new_bs->block_num );
```

Comment:
it seems like `controller::commit_block()` will emit an `accepted_block_header` signal making this redundant